### ISSUE TITLE
[Dark savior] Affichage incorrect du sound test

### ISSUE THREAD

**Issue by fafling** [FCare/Kronos#893]
En WIP du 19/05, et en v2.1.2 publique, le sound test ne s'affiche pas correctement dans les 2 noyaux :
- En OpenGL, il y a une semi-transparence incorrecte entre les sprites (le texte) et le NBG0 :
![image](https://user-images.githubusercontent.com/7624262/82761399-85c6bb00-9dfa-11ea-8532-744e859a4e04.png)

- En CS, l'affichage des sprites est corrompu :
![image](https://user-images.githubusercontent.com/7624262/82761813-8ad93980-9dfd-11ea-982c-d20e0c63bf5c.png)

Affichage correct dans Kronos 1.7.0 :
![image](https://user-images.githubusercontent.com/7624262/82762010-b1e43b00-9dfe-11ea-9770-67c68aca0c77.png)

L'affichage a toujours été corrompu en noyau CS.

La semi-transparence du noyau OpenGL ne peut pas avoir lieu sur la console car le jeu est en haute résolution et utilise le color ram mode 1. Dans ces conditions, la color calculation ne fonctionne que si le pixel de la 2nd image est en format RGB (table 12.1 p. 236 du mode d'emploi du VDP2). Or les sprites et le NBG0 sont en mode palette.
![image](https://user-images.githubusercontent.com/7624262/82762089-4058bc80-9dff-11ea-9afb-73a704ac0e68.png)
![image](https://user-images.githubusercontent.com/7624262/82762127-8877df00-9dff-11ea-9661-fff22161784a.png)

Save state :
[dark_savior_sound_test.zip](https://github.com/FCare/Kronos/files/4674157/dark_savior_sound_test.zip)

Testé avec le CD US.

**Comment by BenjaminSiskoo** [comment]
Kronos 2.1.5 du 05.01.2021 : même problème

**Comment by BenjaminSiskoo** [comment]
Kronos 2.2.0 du 15.12.2021
Mode CS

Sur la version Japonaise de Dark Savior, on n’aperçois même plus les options.

![image](https://user-images.githubusercontent.com/12481242/146164747-065395de-43d4-4f84-879f-6f8f1b7f6ff2.png)

**Comment by fafling** [comment]
Oui, bizarre cette différence d'affichage entre les régions. La version US a toujours l'affichage des captures de mon 1er commentaire. Pas vu de différence entre les paramètres des sprites sur le VDP2.

Une bizarrerie sur les commandes des sprites : le Gouraud shading est activé alors qu'il n'est pas fonctionnel sur un framebuffer 8 bit (c'est un framebuffer haute résolution).
![image](https://user-images.githubusercontent.com/7624262/146282261-2b83e028-8316-40fe-8e01-c619701cfbeb.png)

**Comment by BenjaminSiskoo** [comment]
Kronos 2.3.1 du 29.11.2022
Mode Cs

La version japonaise a retrouvé le menu entre le test du 15.12.2021 et la version public 2.3.1.

**Comment by fafling** [comment]
Concernant la restriction de la color calculation en high res horizontale en color ram mode 1, l'écran d'option de Dark savior est intéressant car il a les 2 cas de figure :
- La color calculation ne doit pas marcher si la 2ème image est en mode palette, c'est le cas entre le texte (sprites) et l'arrière-plan (NBG0).
- La color calculation doit marcher si la 2ème image est en mode RGB, c'est le cas entre l'arrière-plan (NBG0) et le back screen. Cela permet l'assombrissement de l'arrière-plan.

Un autre exemple de color calculation incorrecte appliquée par Kronos en color ram mode 1 en high res horizontale (démo par Hassmaschine/nando) :
[game (1).zip](https://github.com/user-attachments/files/17289044/game.1.zip)
Affichage incorrect de Kronos v2.6.2 à droite, affichage correct de Mednafen 1.32.1 à gauche :
![image](https://github.com/user-attachments/assets/8ea40d18-c699-4b72-84c3-e76fd65a963c)

**Comment by BenjaminSiskoo** [comment]
Amélioration du sound test sur Dark Savior 

![image](https://github.com/user-attachments/assets/cde5848c-751b-4852-962d-5621a29baf18)
